.changeset/sharp-news-smoke.md

```diff
@@ -0,0 +1,26 @@
+---
+'@backstage/backend-common': minor
+---
+
+Added support for Azure AD authentication for PostgreSQL. To define Azure credentials add `tenantId`, `clientId`, `clientSecret` to `app-config.yaml` file.
+
+```
+backend:
+  database:
+    .....
+    auth:
+      azure:
+        tenantId: 'your_tenantId'
+        clientId: 'your_clientId'
+        clientSecret: 'your_clientSecret'
+```
+
+You can also use other methods to define credentials. See more about [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#defaultazurecredential)
```

Comment:
Should all of this info be in the docs somewhere instead of in the changeset where it'll get lost over time? Trouble is, there's no great place to put it yet it seems ... could possibly have been in the integrations/azure section (in a new page next to https://backstage.io/docs/integrations/azure/locations for example), but not sure 🤔 

Maybe skip for now, but instead putting a properly commented section in the config schema as per above to cover our bases until we have a better docs structure.

Reply:
There is documentation on how to configure the plugin database. Could these changes go there for now? https://backstage.io/docs/tutorials/configuring-plugin-databases/#configuration

Reply:
We have https://backstage.io/docs/integrations/azure/org#authenticating-with-microsoft-graph for microsoft graph.  We could re-use parts of that.

Would also need a bit of unification around all the AAD auth implementations - I think most support the `DefaultAzureCredential` setup, but in most other cases the app-config.yml configuration supported is a bit all over the place, which goes well beyond the scope of this PR.